index.js

```diff
@@ -1,5 +1,122 @@
 'use strict';
 
+const VersionChecker = require('ember-cli-version-checker');
+const MergeTrees = require('broccoli-merge-trees');
+const Funnel = require('broccoli-funnel');
+
+const DEFAULT_OPTIONS = {
+  includeScriptTags: true,
+  legacyTargets: null,
+  evergreenTargets: [
+    'last 2 Chrome versions',
+    'last 2 Firefox versions',
+    'last 2 Safari versions',
+    'last 2 Edge versions',
+    'last 2 ChromeAndroid versions',
+    'last 2 iOS versions',
+  ],
+};
+
 module.exports = {
-  name: require('./package').name
+  name: require('./package').name,
+
+  included() {
+    this._super.included(...arguments);
+
+    let appOptions = (this.app.options || {})['ember-cli-babel-polyfills'];
+
+    this.options = Object.assign({}, DEFAULT_OPTIONS, appOptions);
+
+    this._isBabel7 = new VersionChecker(this.project).for('ember-cli-babel').gte('7.0.0');
+
+    this.app.import('vendor/ember-cli-babel-polyfills/shared.js', {
+      outputFile: 'assets/polyfill-shared.js'
+    });
+
+    this.app.import('vendor/ember-cli-babel-polyfills/legacy.js', {
+      outputFile: 'assets/polyfill-legacy.js'
+    });
+
+    this.app.import('vendor/ember-cli-babel-polyfills/evergreen.js', {
+      outputFile: 'assets/polyfill-evergreen.js'
+    });
+  },
+
+  treeForVendor() {
+    let Rollup = require('broccoli-rollup');
+    let resolve = require('rollup-plugin-node-resolve');
+    let commonjs = require('rollup-plugin-commonjs');
+
+    let writeFile = require('broccoli-file-creator');
+    let TransformAmd = require('./lib/transform-amd');
+
+    let legacyTargets = this.options.legacyTargets || this.project.targets;
+    let evergreenTargets = this.options.evergreenTargets
+
+    let entries = new MergeTrees([
+      writeFile('legacy.js', this._getEntryForTargets(legacyTargets)),
+      writeFile('evergreen.js', this._getEntryForTargets(evergreenTargets)),
+    ]);
+
+    let rolledUp = new Rollup(entries, {
+      rollup: {
+        experimentalCodeSplitting: true,
+        input: ['legacy.js', 'evergreen.js'],
+        output: {
+          dir: 'output',
+          format: 'amd',
+        },
+        plugins: [
+          resolve(),
+          commonjs(),
+        ],
+      },
+    });
+
+    return new Funnel(new TransformAmd(rolledUp), {
+      srcDir: 'output',
+      destDir: 'ember-cli-babel-polyfills',
+      getDestinationPath(path) {
```

Comment:
I think you shouldn’t need this? You should be able to control the shared files name via roll up options

Reply:
Two problems:

1. Couldn't find any documentation for how to do it except [this long thread](https://github.com/rollup/rollup/issues/1995) that seemed to indicate you have to use plugins
2. `broccoli-rollup` is pinned to an old version of Rollup at the moment. Would need to be updated, and since this is time sensitive I figured this was a decent solution for the time being.